File: functions/alsa-lib/variants/go/1_iec958_parity_7.go
package main

func iec958_parity(data uint32) uint32 {
	data >>= 4
	parity := uint32(0)
	for i := 0; i < 27; i++ {
		if (data & 1) != 0 {
			parity++
		}
		data >>= 1
	}
	return parity & 1
}


func main(){
    iec958_parity(0)
}

File: functions/alsa-lib/variants/c/1_iec958_parity_7.c
#include <stdint.h>

static unsigned int iec958_parity(unsigned int data)
{
	unsigned int parity = 0;
	data >>= 4;
	for (int bit = 0; bit < 27; bit++) {
		parity += (data >> bit) & 1;
	}
	return parity & 1;
}


int main(){
    iec958_parity(0);
}
